bindings/azure/eventhubs/metadata.yaml

```diff
@@ -0,0 +1,165 @@
+# yaml-language-server: $schema=../../../component-metadata-schema.json
+schemaVersion: v1
+type: bindings
+name: azure.eventhubs
+version: v1
+status: stable
+title: "Azure Event Hubs"
+urls:
+  - title: Reference
+    url: https://docs.dapr.io/reference/components-reference/supported-bindings/eventhubs/
+binding:
+  input: true
+  output: true
+  operations:
+    - name: create
+      description: "Create an event subscription."
+capabilities: []
+authenticationProfiles:
+  - title: "Connection string"
+    description: "Authenticate using a connection string."
+    metadata:
+      - name: connectionString
+        required: true
+        sensitive: true
+        description: |
+          Connection string for the Event Hub or the Event Hub namespace.
+          Mutually exclusive with eventHubNamespace field.
+          Required when not using Azure AD Authentication.
+        example: '""Endpoint=sb://{EventHubNamespace}.servicebus.windows.net/;SharedAccessKeyName={PolicyName};SharedAccessKey={Key};EntityPath={EventHub}""'
+      - name: eventHub
+        type: string
+        required: false # Optional when a connectionString is provided
+builtinAuthenticationProfiles:
+  - name: "azuread"
+    metadata:
+      # Either connectionString or eventHubNamespace must be defined, but not both
+      - name: eventHubNamespace
+        type: string
+        required: true
+        description: |
+          The Event Hub Namespace name.
+          Mutually exclusive with connectionString field.
+          Required when using Azure AD Authentication.
+        example: '"namespace"'
+      - name: eventHub
+        type: string
+        required: true # required for AD
+        description: |
+          The name of the Event Hubs hub (“topic”). Required if using Azure AD
+          authentication or if the connection string doesn’t contain an
+          EntityPath value.
+        example: '"mytopic"'
+metadata:
+  # Input and output metadata
+
+  # Entity management is only possible when using Azure AD
+  - name: enableEntityManagement
+    type: bool
+    required: false
+    default: "false"
+    description: |
+      Boolean value to allow management of the EventHub namespace and
+      storage account.
+  - name: messageRetentionInDays
+    type: number
+    required: false
+    default: "1"
+    example: "90"
+    description: |
+      Number of days to retain messages for in the newly created Event
+      Hub namespace. Used only when entity management is enabled.
+  - name: partitionCount
+    type: number
+    required: false
+    default: "1"
+    example: "3"
+    description: |
+      Number of partitions for the new Event Hub namespace. Used only when
+      entity management is enabled.
+  - name: partitionId
```

Comment:
I think it would be nice adding `alias`, `deprecated` (or `deprecatedSince`) and perhaps even a `seeAlso` or `related` (although the latter might be a bit controversial).

Would that require a proposal/issue or is this something that doesn't require that much formality?

Reply:
I think `deprecated` for a field would be useful. I can see `related` potentially useful too.

I don't think I'd like to highlight aliased fields. Normally when a field is aliased it's because the old one had a "bad name"  (or some other reason) and we shouldn't encourage using that anyways.

Reply:
Ok. One potential that I see with explicitly marking the aliased fields and naming the aliases is that we could potentially, when parsing, mark a configuration that defines multiple variants of an aliased field as a configuration error. Marking them as deprecated alone would not prevent this sort of situation nor allow for this sort of validation.

Reply:
Depends on https://github.com/dapr/components-contrib/pull/2794

Reply:
The PR above was merged and deprecated fields were marked as such. Resolving.